internal/xds/translator/translator.go

```diff
@@ -637,6 +644,43 @@ func (t *Translator) addRouteToRouteConfig(
 	return errs
 }
 
+// SortVirtualHosts sorts the virtual hosts by domain so the order is deterministic.
+func sortVirtualHosts(vHosts []*routev3.VirtualHost) {
+	slices.SortFunc(vHosts, func(a, b *routev3.VirtualHost) int {
+		if a == nil && b == nil {
+			return 0
+		}
+		if a == nil {
+			return 1
+		}
+		if b == nil {
+			return -1
+		}
+
+		domainA := a.Domains[0]
+		domainB := b.Domains[0]
+
+		// Wildcards are less specific, so they should come after exact matches
+		aIsWildcard := strings.HasPrefix(domainA, "*")
+		bIsWildcard := strings.HasPrefix(domainB, "*")
+
+		if aIsWildcard && !bIsWildcard {
+			return 1 // a (wildcard) is less specific, comes after b
+		}
+		if !aIsWildcard && bIsWildcard {
+			return -1 // a (exact) is more specific, comes before b
+		}
+
+		// If both are wildcards, or both are exact matches, compare their lengths
+		if len(domainA) != len(domainB) {
+			return len(domainB) - len(domainA) // longer comes first
+		}
```

Comment:
Would it be useful to include a comparison of root domains before this?
For example with the below two domains, which is preferred to come first?
* b.example.a.com
* a.example.b.com